packages/filesystem/src/common/filesystem-utils.ts

```diff
@@ -18,24 +18,46 @@ import { FileStat } from '../common/files';
 import URI from '@theia/core/lib/common/uri';
 
 export namespace FileSystemUtils {
+    export const FILE_NAME_SEPARATOR = ' ';
 
     /**
      * Generate unique URI for a given parent which does not collide
      *
-     * @param parentUri the `URI` of the parent
      * @param parent the `FileStat` of the parent
-     * @param name the resource name
-     * @param ext the resource extension
+     * @param targetUri the initial URI
+     * @param suffix an optional string to append to the file name, in case of collision (e.g. `copy`)
      */
-    export function generateUniqueResourceURI(parentUri: URI, parent: FileStat, name: string, ext: string = ''): URI {
+    export function generateUniqueResourceURI(parent: FileStat, targetUri: URI, isDirectory: boolean, suffix?: string): URI {
```

Comment:
Please add a breaking change entry in the changelog for this.

Reply:
I updated the changelog